.github/workflows/test-push.yml

```diff
@@ -148,3 +148,53 @@ jobs:
         with:
           name: Reproducible Build ${{ matrix.os }} (${{ matrix.java }}) Results
           path: build/reports/
+
+  # Special case this test to run on Java 21
```

Comment:
Note (mostly for myself): this can be removed in 1.7 since [the Fabric API test requires it](https://github.com/FabricMC/fabric-loom/commit/097fd98fc91fdbe9ec42f2304991b68bb29751e4)